query-connector/src/app/query/components/CustomizeQuery.tsx

```diff
@@ -46,41 +51,35 @@ const CustomizeQuery: React.FC<CustomizeQueryProps> = ({
   goBack,
 }) => {
   const [activeTab, setActiveTab] = useState<DibbsValueSetType>("labs");
-  const { labs, conditions, medications } =
-    mapValueSetsToValueSetTypes(queryValuesets);
-  const [valueSetOptions, setValueSetOptions] = useState({
```

Comment:
Setting the state this way was causing the tests to not re-render the component when the value for `queryValuesets` updated. As a result, even when the data came back from the backend, the page was rendering as if nothing was updating. 

Rewriting this to use a `useEffect` instead guarentees that React with reset the state correctly when the value of `queryValuesets` updates

Reply:
This makes perfect sense and is a prime example of why, not a front-end person, know _just_ enough react to be an absolute hailstorm of danger to all proper stateful patterns 😅